docs/dev/ui_style_guide.rst

```diff
@@ -69,8 +69,13 @@ try to adhere to the following principles:
 
     <template>
       <j-tray-plugin
-        :description="docs_description || 'Plugin description.'"
-        :link="docs_link || 'https://jdaviz.readthedocs.io/en/'+vdocs+'/'+config+'/plugins.html#plugin-name'"
+        description='Plugin description.'
+        :link="'https://jdaviz.readthedocs.io/en/'+vdocs+'/'+config+'/plugins.html#plugin-name'"
+        :irrelevant_msg="irrelevant_msg"
```

Comment:
Maybe add explanation on when devs should consider something "irrelevant" or not. Do we still need "disabled" then? What are the differences when devs decide which one to use?

Reply:
The distinction with disabled is meaningful. "irrelevant" extends upon the “disabled” plugin feature, which marks a plugin as disabled with a message in the UI, but does not “remove” it from the plugin tray or API. "irrelevant" additionally removes plugins from the UI and public API.

Reply:
So once it is marked as irrelevant, can it become relevant again later in the same session? I think this was discussed before but I don't remember and none of those discussions are documented in the diff.

Is the only use for this to hide a WIP plugin?

Reply:
> Is the only use for this to hide a WIP plugin?

It can have more general uses. We could also use it to put plugins away depending on the state of the app or viewer.

Reply:
This is a traitlet, so we can respond to anything we want to toggle its "relevancy", yes.  Disabled leaves the plugin in the tray but just replaces the UI with a message (which is useful for cases where you want to tell the user WHY something disappeared).  Imagine merging specviz and cubeviz, its obvious that the slice plugin is irrelevant without cube data, but hiding spectral analysis because of a unit conversion switch probably deserves showing a message.  We can clarify a pattern for when to use each as they are used more often - and the under-the-hood implementation is very similar, so should be easy to migrate from one to the other when warranted.

Reply:
(I'm not really sure where to document this in the docs and I think it'll be a bit liquid for a while until we get a clear idea of how we want to use this... for now, there at least is a good papertrail in this discussion - and the PR description, so thanks for brining it up)

Reply:
Can we clarify the pattern now as you envisioned and clean it up later? We cannot guarantee new devs would be able to dig up this conversation here. A few sentences in the dev docs at least would be searchable in RTD.

Reply:
ok, done